### PR TITLE
DM-13843

### PR DESCRIPTION
**Pull request by kfindeisen**
Here's a first draft of the Development Model section of the paper. Unfortunately, while I can describe what we do, I had to guess at which advantages/disadvantages actually motivated our workflow (the only sources I could find were the Developer Guide and RFC-21). So the text will probably require a fair bit of editing by somebody with a more manager-y perspective.

### CODE REVIEW

File: git.tex
Comment (timj) [inline git.tex]: Can we use one sentence per line please?

File: git.tex
Comment (timj) [inline git.tex]: Point to the https://github.com/lsst location.

File: git.tex
Comment (timj) [inline git.tex]: I think this relates to us keeping branches around and not deleting them. With forks the branch never turns up in the parent repo. It's easier to audit the work later if the branch and the Jira ticket are related: you can work it out if the branch was deleted but it depends on you parsing the merge commits looking for the branch name. @ktlim has suggested that deleting merged branches is okay so long as we replace them with tags. Also, Jira can't see branches on forks but can see branches on the parent repo (it may be that we don't care about that but it is sometimes useful to be able to go to a branch directly from Jira even if no PR has been made yet).

File: git.tex
Comment (timj) [inline git.tex]: Now that branch protections are in place we can merge with the green button.

File: git.tex
Comment (timj) [inline git.tex]: (whilst ensuring `--no-ff` is used).

File: git.tex
Comment (timj) [inline git.tex]: It also means that all the work associated with the Jira story is on the branch itself. Once you allow merges without rebasing you end up with code changes in the commit that are part of the story but which are difficult to disentangle from other work that was merged before you merged.

File: git.tex
Comment (timj) [inline git.tex]: or merge without forgetting to us `--no-ff` or merge without remembering to rebase. We also want to avoid those fake commits caused by the remote changing whilst you have updated master locally.

File: git.tex
Comment (timj) [inline git.tex]: We have historically allowed direct commits to master of documentation fixes but now that doc fixes can cause flake8 failures we no longer allow this.

File: other-workflows.tex
Comment (timj) [inline other-workflows.tex]: Although, as we approach commissioning we may well start to adopt release branches. We do currently use release branches for the major six monthly releases but they have very few commits on them and only those needed to fix major problems during the release process.

File: other-workflows.tex
Comment (timj) [inline other-workflows.tex]: We choose a weekly at some point to make a full release (Which comes with a characterization report).

**Review by timj — APPROVED** [review]
Some minor comments. Happy to merge when you've dealt with them. Thanks.

File: git.tex
Comment (kfindeisen) [reply git.tex]: Perhaps we could, but our [internal docs](https://developer.lsst.io/work/flow.html#merging) say not to.

Comment (timj) [reply git.tex]: Yes. Need to fix. Branch protections let us do that without breaking policy. I thought I had made that change when I updated the branch protection docs in developer guide (and I demonstrated to @ktlim that it worked) but I obviously need to edit the dev guide.

File: git.tex
Comment (kfindeisen) [reply git.tex]: I guess I'm not sure exactly what "audit the work" means, since it's not something I see.

Comment (timj) [reply git.tex]: I don't think anyone has ever asked to audit a story. It is the case that Jira won't see forks and we want Jira to see the branches.

File: git.tex
Comment (kfindeisen) [reply git.tex]: I'd like to avoid including command line details -- as I understand it this section should be about our process, not what a developer needs to do to make it happen. Given that I mention the trivial merge commits later in the paragraph, is it necessary to bring up the flag itself?

File: git.tex
Comment (kfindeisen) [reply git.tex]: Are you asking for me to include an aside about documentation as opposed to code?

File: git.tex
Comment (kfindeisen) [reply git.tex]: Per your comment on `#dm`, should this be a full reference? It seems a bit weird.

File: git.tex
Comment (timj) [reply git.tex]: I'm talking about fixes to docstrings in code, so I think it's worth mentioning that we used to allow them direct to master but now require PRs because they can break tests.

File: other-workflows.tex
Comment (kfindeisen) [reply other-workflows.tex]: That is typically what release branches are supposed to look like.

In any case, my knowledge of LSST's release process is minimal (and apparently wrong), so I don't think I can write that bit. I can add a TODO for somebody else to do it.

File: git.tex
Comment (timj) [reply git.tex]: I am suggesting to make clear that fast forward merges are not allowed.

File: git.tex
Comment (timj) [reply git.tex]: I don't think so. It's not really something citable. It's a link to the LSST repositories, not an article or guide.

**Pull request merged**
